project/test_cases.py

```diff
@@ -3,46 +3,130 @@
+            assigned_4 = student.session_4_matched
+
+            if assigned_1 == False:
+                assert False
+            elif assigned_2 == False:
+                assert False
+            elif assigned_3 == False:
+                assert False
+            elif assigned_4 == False:
+                assert False
+            else: 
+                print "Students Assigned"
+    
+    def test_is_student_in_session(self):
+        """Tests to see if a student is in the session that it is assigned to"""
+        students = SESSION.query(Student).all()
+        sess1 = SESSION.query(Project).filter_by(session_number=1).all()
+        sess2 = SESSION.query(Project).filter_by(session_number=2).all()
+        sess3 = SESSION.query(Project).filter_by(session_number=3).all()
+        sess4 = SESSION.query(Project).filter_by(session_number=4).all()
+        for student in students:
+            for project in sess1:
+                print "yes"
+                has_student = 0
+                studentlist = project.students
+                if student in studentlist:
+                    has_student += 1
+                if has_student != 1:
+                    assert False
+                else:
+                    print "Session 1 has student"
+
+            for project in sess2:
+                print "yes"
+                has_student = 0
+                studentlist = project(students)
+                if student in studentlist:
+                    has_student += 1
+                if has_student != 1:
+                    assert False
+                else: 
+                    print "Session 2 has student"
+
+            for project in sess3:
+                print "yes"
+                has_student = 0
+                studentlist = project(students)
+                if student in studentlist:
+                    has_student += 1
+                if has_student != 1:
+                    assert False
+                else:
+                    print "Session 3 has student"
+
+            for project in sess4:
+                print "yes"
+                has_student = 0
+                studentlist = project(students)
+                if student in studentlist:
+                    has_student += 1
+                if has_student != 1:
+                    assert False
+                else:
+                    print "Session 4 has student"
+        print "hello world"
+
+    def test_most_students_get_first(self):
+        """Tests to see if most students get their first pref"""
+        print "this is where the fun begins"
+        numpref = 0
+        first_pref = SESSION.query(Pref).filter_by(pref_number = 1).all()
+        print "asadf"
+        all_students = SESSION.query(Student).count()
+        print "ill have you know i graduated at the top of my class"
+        for pref in first_pref:
+            student = pref.student
+            projects = SESSION.query(Project).all()
+            for project in projects:
```

Comment:
Why are you looping through all the projects? 

You shouldjust use the project the name in the preference to grab the project from the database and check to see if the student is in the project's student list